### ISSUE TITLE
send() ignores interface for some IPv6 packets

### ISSUE THREAD

**Issue by chkr-private** [secdev/scapy#4013]
### Brief description

When sending multicast or link-local IPv6 L3 packets, the supplied interface is ignored by send() on a host with multiple interfaces.

Additionally, when using a workaround with sendp(), the source MAC and IP are not set according to the sending interface. (might be a different problem, please advice if there should be a 2nd Bug Report).

### Scapy version

2.4.5 as well as  9d81d48812a5adb4e35bb9672a1a7fba9ec94a2c (current master)

### Python version

3.10.11

### Operating system

Fedora 36

### Additional environment information

It was quite unexpected that the "iface" parameter of send() was silently ignored when dealing only with L3 packets.

Other tools (e.g. ping6) let the user specify the outgoing interface using the scope ID syntax ('fe80:...:1234%eth2', that didn't work in scapy) and chose the correct source IP/MAC accordingly. Additionally, they also warn the user when the interface can't be unambiguously determined.

I tried to dig into scapy's source a little bit, but I'm not sure I'm able to provide a fix without some pointers. Please let me know your thoughts and/or provide some hints.

### How to reproduce

1. try to send an ICMPv6 echo request via the 2nd ethernet interface of the host

```
>>> p = IPv6(dst='FF02::1')/ICMPv6EchoRequest()
>>> send(p, iface='eth1')
```

3. 2nd attempt, specify the src address of the 2nd interface "eth1" explicitly

```
>>> p = IPv6(src='fe80::xxxx:xxff:fexx:xxxx', dst='ff02::1')/ICMPv6EchoRequest()
>>> send(p, iface='eth1')
```

3. use L2 frames

```
>>> p = Ether()/IPv6(dst='ff02::1')/ICMPv6EchoRequest()
>>> sendp(p, iface='eth1')
```

4. use L2 frames, specify source IP

```
>>> p = Ether())/IPv6(src='fe80::xxxx:xxff:fexx:xxxx', dst='ff02::1')/ICMPv6EchoRequest()
>>> sendp(p, iface='eth1')
```

5. use L2 frames, specify source MAC and source IP

```
>>> p = Ether(src='xx:xx:xx:xx:xx:xx'))/IPv6(src='fe80::xxxx:xxff:fexx:xxxx', dst='ff02::1')/ICMPv6EchoRequest()
>>> sendp(p, iface='eth1')
```

### Actual result

1. packet is sent without an error message to the wrong interface (it is sent to the first interface instead of "eth1")
   - same happens when using a link-local destination address instead of the multicast address
2. still sent to the wrong interface
3. packet is sent to the correct interface, but both, the src IPv6 address as well as the src MAC address are the one from the first ethernet interface
4. packet is sent to he correct interface, but the src MAC is still the one from the first interface on the host
5. the outgoing interface, src MAC and src IP are now correct

### Expected result

1. and 2.:  use the supplied interface to send out the packet
  - if there is no interface specified and the destination does not allow to unambiguously infer the outgoing interface, issue a warning

3. and 4.:  src IP+MAC should set according to the supplied outgoing interface

**Comment by guedou** [comment]
TLDR, you should use `sendp()` and use `conf.iface`.

IPv6 link-local addresses are tricky as they exist on all network interfaces. When injecting packets at L3, Scapy will find the best route, and use the corresponding interface to inject the packet.

Concerning attempts 4 and 5, you can force Scapy to use specific source addresses by specifying the value of `conf.iface`.
